src/lib.rs

```diff
@@ -1653,6 +1653,96 @@ impl Bindings {
             ))
         }
     }
+    
+    /// Write these bindings as source text to stdout.
+    pub fn write_to_stdout(&self) -> io::Result<()> {
+        let bindings = self.to_string();
+        
+        match self.rustfmt_generated_string(bindings) {
+            Ok(Some((rustfmt_bindings, rustfmt_stderr))) => { 
+                println!("{}", rustfmt_bindings);
+                eprintln!("{}", rustfmt_stderr);
+            },
+            Ok(None) => (),
+            Err(err) => println!("{:?}", err),
+        }
+        
+        Ok(())
+    }
+    
+    /// Checks if rustfmt_bindings is set and runs rustfmt on the string
+    fn rustfmt_generated_string(&self, source: String) -> io::Result<Option<(String, String)>> {
+        let _t = self.context.timer("rustfmt_generated_string");
+
+        if !self.context.options().rustfmt_bindings {
+            return Ok(None);
+        }
+
+        let rustfmt = if let Ok(rustfmt) = which::which("rustfmt") {
+            rustfmt
+        } else {
+            return Err(io::Error::new(
+                io::ErrorKind::Other,
+                "Rustfmt activated, but it could not be found in global path.",
+            ));
+        };
+        
+        let mut cmd = Command::new(rustfmt);
+        cmd
+            .args(&["--write-mode=display"])
+            .stdin(Stdio::piped())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped());
+            
+        if let Some(path) = self.context
+            .options()
+            .rustfmt_configuration_file
+            .as_ref()
+            .and_then(|f| f.to_str())
+        {
+            cmd.args(&["--config-path", path]);
+        }
+        
+        let mut child = cmd.spawn().expect("Failed to spawn rustfmt.");
+        let mut child_stdin = child.stdin.take().unwrap();
+        let mut child_stdout = child.stdout.take().unwrap();
+        let mut child_stderr = child.stderr.take().unwrap();
+    
+        // Write to stdin in a new thread, so that we can read from stdout on this
+        // thread. This keeps the child from blocking on writing to its stdout which
+        // might block us from writing to its stdin.
+        let stdin_handle = ::std::thread::spawn(move || {
+            child_stdin.write_all(source.as_bytes())
+        });
+    
+        // Read stderr on a new thread for similar reasons.
+        let stderr_handle = ::std::thread::spawn(move || {
+          let mut output = vec![];
+          io::copy(&mut child_stderr, &mut output)
+              .map(|_| String::from_utf8_lossy(&output).to_string())
+        });
+
+        let mut output = vec![];
+        io::copy(&mut child_stdout, &mut output)
+            .expect("Should copy stdout into vec OK");
+            
+        // Ignore actual rustfmt status because it is often non-zero for trivial
+        // things.
+        let _ = child.wait().expect("should wait on rustfmt child OK");
```

Comment:
This was fine for tests, but now that this is for a wider audience, lets do what `rustfmt_generated_file` is doing with the status code:

```rust
                match output.status.code() {
                    Some(0) => Ok(()),
                    Some(2) => Err(io::Error::new(
                        io::ErrorKind::Other,
                        format!("Rustfmt parsing errors:\n{}", stderr),
                    )),
                    Some(3) => {
                        warn!(
                            "Rustfmt could not format some lines:\n{}",
                            stderr
                        );
                        Ok(())
                    }
                    _ => Err(io::Error::new(
                        io::ErrorKind::Other,
                        format!("Internal rustfmt error:\n{}", stderr),
                    )),
                }
```

You can see more information about `rustfmt`'s exit codes with `rustfmt --help`:

```
$ rustup run nightly rustfmt -- --help

<snip>

Exit Codes:
    0 = No errors
    1 = Encountered operational errors e.g. an IO error
    2 = Failed to reformat code because of parsing errors
    3 = Code is valid, but it is impossible to format it properly
    4 = Formatted code differs from existing code (write-mode diff only)
```